src/DeviceInterface/ExtensionInterface.js

```diff
@@ -27,31 +27,56 @@ class ExtensionInterface extends InterfacePrototype {
             testMode: this.isTestMode()
         }
         const tooltipKinds = {
-            [POPUP_TYPES.EmailProtection]: 'legacy',
-            [POPUP_TYPES.EmailSignup]: 'emailsignup'
+            [TOOLTIP_TYPES.EmailProtection]: 'legacy',
+            [TOOLTIP_TYPES.EmailSignup]: 'emailsignup'
         }
-        const tooltipKind = tooltipKinds[this.getShowingTooltip()] || tooltipKinds[POPUP_TYPES.EmailProtection]
+        const tooltipKind = tooltipKinds[this.getShowingTooltip()] || tooltipKinds[TOOLTIP_TYPES.EmailProtection]
 
         return new HTMLTooltipUIController({ tooltipKind, device: this }, htmlTooltipOptions)
     }
 
-    get hasDismissedEmailSignup () {
-        // TODO -- implement peristed dismissed timestamp
-        return false
-    }
-
     getShowingTooltip () {
         if (this.hasLocalAddresses) {
-            return POPUP_TYPES.EmailProtection
+            return TOOLTIP_TYPES.EmailProtection
         }
 
-        if (this.settings.featureToggles.emailProtection_incontext_signup && !this.hasDismissedEmailSignup) {
-            return POPUP_TYPES.EmailSignup
+        if (this.settings.featureToggles.emailProtection_incontext_signup && this.settings.incontextSignupDismissed === false) {
+            return TOOLTIP_TYPES.EmailSignup
         }
 
         return null
     }
 
+    onIncontextSignupDismissed () {
+        this.settings.setIncontextSignupDismissed(true)
+        chrome.runtime.sendMessage({
```

Comment:
We probably want to move this call to the transport and use `deviceApiCalls` like Mateusz is doing [here](https://github.com/duckduckgo/duckduckgo-autofill/pull/242/files#diff-ca1fc5ddd4bcb9795968042de279c2f8b972cc2991f6d8e40bce844879841583R70).